src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/RazorCompletionResolveData.cs

```diff
@@ -30,21 +30,22 @@ public static RazorCompletionResolveData Unwrap(CompletionItem completionItem)
         return context;
     }
 
-    public static void Wrap(VSInternalCompletionList completionList, TextDocumentIdentifier textDocument, bool supportsCompletionListData)
+    public static void Wrap(VSInternalCompletionList completionList, TextDocumentIdentifier textDocument, VSInternalClientCapabilities clientCapabilities)
     {
         var data = new RazorCompletionResolveData(textDocument, OriginalData: null);
 
-        if (supportsCompletionListData)
+        if (clientCapabilities.SupportsAnyCompletionListData())
         {
-            if (completionList.Data is not null)
+            if (clientCapabilities.SupportsCompletionListData() || completionList.Data is not null)
```

Comment:
>completionList.Data is not null [](http://example.com/codeflow?start=67&length=31)

Is it valid for Data to not be null but SupportsCompletionListData to be true?

Reply:
"Valid" is a funny word when it comes to LSP and serialization. If a client said they didn't support Data on the completion list, then presumably it would have no use as they would never read it, but wouldn't be illegal for someone to send it across the wire. Extra JSON data is not an error in any scenario I've ever seen.

The extra check here is essentially just paranoia: It would take a bad server (sending data back when the client says they don't want it) and a bad client (reading data when they said they didn't), but at the end of the day if this method doesn't wrap every bit of data it sees, its a bug waiting to happen. And we hit that bug in lots of our tests, which essentially act as LSP clients, but are exceedingly badly written as far as LSP clients go :)